sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala

```diff
@@ -212,11 +210,11 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
 
         public InternalRow next() {
           currentRow += 1;
-          bufferHolder.reset();
+          rowWriter.reset();
           rowWriter.zeroOutNullBytes();
           ${extractorCalls}
-          unsafeRow.setTotalSize(bufferHolder.totalSize());
-          return unsafeRow;
+          rowWriter.setTotalSize();
+          return rowWriter.getRow();
```

Comment:
Is there any place where we call `getRow()` without calling `setTotalSize()` before that? If there aren't then I'd combine the two.

Reply:
Should we call `reset()` and `setTotalSize()` as the interpreted version does?